test/test_upnp.cpp

```diff
@@ -176,12 +176,12 @@ void run_upnp_test(char const* root_filename, char const* router_model, char con
 			ec.clear();
 			break;
 		}
-		if (upnp_handler->router_model() != "") break;
+		if (!upnp_handler->router_model().empty()) break;
 		test_sleep(100);
 	}
 
 	std::cerr << "router: " << upnp_handler->router_model() << std::endl;
-	TEST_EQUAL(upnp_handler->router_model(), router_model);
+	//TEST_EQUAL(upnp_handler->router_model(), router_model);
```

Comment:
the idea is that we should pick up the fake UPnP device the test spins up. I take it you find this unreliable, is that right?

Reply:
Exactly. The problem here is that ```lt::upnp``` constructor ignores interface passed to it and thus upnp instance will update ```m_model``` each time it receives upnp xml. Since it is neither possible to predict order in which routers are discovered nor to select router by the interface it makes no sense to expect ```upnp_handler->router_model()``` to return fake device name.
I beleive ```TEST_CHECK(!upnp_handler->router_model().empty());``` might be used instead. What do you think?

Reply:
that seems reasonable, yes. (at least until the upnp implementation can stick to a specific interface)